productsManagement/src/main/java/com/productsmanagement/service/ProductManagementServiceImpl.java

```diff
@@ -1,24 +1,46 @@
 package com.productsmanagement.service;
 
-import com.productsmanagement.exception.DuplicatedProductException;
-import com.productsmanagement.util.Utils;
 import com.skytouch.dto.ProductDTO;
+import com.skytouch.utils.Constants;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
+import java.util.function.Supplier;
 
 @Service
 public class ProductManagementServiceImpl implements ProductManagementService {
+    private final RabbitTemplate rabbitTemplate;
+
+    @Autowired
+    public ProductManagementServiceImpl(RabbitTemplate rabbitTemplate) {
+        this.rabbitTemplate = rabbitTemplate;
+    }
+
     @Override
     public List<ProductDTO> getAllProducts() {
-        //TODO: Implements to retrieve from queue
-        return Utils.getMockProducts();
+        return convertSendAndReceiveOrThrow(
+                () -> rabbitTemplate.convertSendAndReceive(
+                        Constants.GET_PRODUCT_QUEUE,
+                        Constants.GET_PRODUCT_QUEUE));
     }
 
     @Override
     public ProductDTO addProduct(ProductDTO productDTO) {
-        int a = 1/0;
-        //TODO: Implements to add product queue
+        String  result = convertSendAndReceiveOrThrow(
+                () -> rabbitTemplate.convertSendAndReceive(
```

Comment:
`RabbitTemplate.convertSendAndReceive()` should be used only to get the list of products, not to add a product.